gwdetchar/io/html.py

```diff
@@ -766,17 +766,31 @@ def fancybox_img(img, linkparams=dict(), lazy=False, **params):
     aparams.update(linkparams)
     img = str(img)
     substrings = os.path.basename(img).split('-')
-    channel = '%s-%s' % tuple(substrings[:2])
-    duration = substrings[-1].split('.')[0]
-    page.a(href=img, id_='a_%s_%s' % (channel, duration), **aparams)
+    try:
+        # If is the expected format, use channel and duration
+        ifo_str = substrings[0]
+        if not (len(ifo_str) == 2 and ifo_str[0].isalpha() and
+                ifo_str[-1].isdigit()):
```

Comment:
Just a quick question: why do you use `ifo_str[-1]` instead of `ifo_str[1]`? It looked like the comment string from your last commit was for something about `len(ifo_str) = 1` case, but doesn't the `len(ifo_str) != 2` case already trigger the `TypeError` exception?

Reply:
This issue is that all of the conditions are checked before evaluating the `if` statement, rather than checking the conditions sequentially. In the case of a 1-length string, `len(ifo_str) == 2` would evaluate to `False`, but  `ifo_str[1].isdigit()` would return an `IndexError`. I could add a catch for this, but decided to just test  `ifo_str[-1].isdigit()`, which would not encounter an error for a 1-length string.

Reply:
Are you sure that python doesn't see the first argument as `False` in the if statement and then automatically moves on to the else? I am almost sure that I've relied on length checking as the first in a list of conditionals and it works. I'll do a quick check with some code...

Reply:
For example:
```python
>>> ifo_str = 'H'
>>> if not (len(ifo_str) == 2 and ifo_str[0].isalpha() and ifo_str[1].isdigit()):
...   print('foo')
... else:
...   print('bar')
... 
foo
>>> if not (len(ifo_str) == 2 and ifo_str[0].isalpha() and ifo_str[1].isdigit()):
...   print('foo')
... else:
...   print('bar')
... 
bar
```
Seems to work for me

Reply:
Thanks for the example @eagoetz, I misunderstood how these compound `if` statements were being tested. I just pushed your suggestion.

Reply:
@dethodav Thanks! It wasn't a big deal, but I had just noticed your changes and wondered if I was missing something. Thanks for checking and making an adjustment that was not strictly required for the review. All good though, thanks for everything!